content/docs/tutorials/acme/http-validation.md

```diff
@@ -108,22 +108,28 @@ verify domain ownership. To verify ownership of each domain mentioned in an
 `http01` stanza, cert-manager will create a Pod, Service and Ingress that
 exposes an HTTP endpoint that satisfies the HTTP01 challenge.
 
-The fields `ingress` and `ingressClass` in the `http01` stanza can be used to
-control how cert-manager interacts with Ingress resources:
+The fields `name`, `ingressClassName`, and `class` in the `http01` stanza can be
+used to control how cert-manager interacts with Ingress resources:
 
-- If the `ingress` field is specified, then an Ingress resource with the same
+- If the `name` field is specified, then an Ingress resource with the same
   name in the same namespace as the Certificate must already exist and it will
   be modified only to add the appropriate rules to solve the challenge.
   This field is useful for the Google Cloud Loadbalancer ingress controller, 
   as well as a number of others, that assign a single public IP address for 
   each ingress resource.
   Without manual intervention, creating a new ingress resource would cause any
   challenges to fail.
-- If the `ingressClass` field is specified, a new ingress resource with a
-  randomly generated name will be created in order to solve the challenge.
-  This new resource will have an annotation with key `kubernetes.io/ingress.class`
-  and value set to the value of the `ingressClass` field.
-  This works for the likes of the NGINX ingress controller.
+- If the `ingressClassName` field is specified, a new ingress resource with a
+  randomly generated name will be created in order to solve the challenge. This
+  new resource will have the field `ingressClassName` with with the value of the
+  `ingressClassName` field. This is the recommended way of configuring which
+  Ingress controller should be used. This works for the likes of the NGINX
+  ingress controller.
+- If the `class` field is specified, a new ingress resource with a randomly
+  generated name will be created in order to solve the challenge. This new
+  resource will have an annotation with key `kubernetes.io/ingress.class` and
+  value set to the value of the `class` field. This field is only recommended
+  with ingress-gce which doesn't support the `ingressClassName` field.
```

Comment:
Could you add a link to a reference about ingress-gce not supporting this field?

Reply:
I added a link to https://cloud.google.com/kubernetes-engine/docs/how-to/load-balance-ingress. I wasn't able to pin-point the link to the exact location. Here is the relevant bit:

> There are two GKE Ingress classes. To specify an Ingress class, you must use the kubernetes.io/ingress.class annotation. You cannot specify a GKE Ingress using spec.ingressClassName.